tests/test_docker-compose.sh

```diff
@@ -0,0 +1,660 @@
+docker-compose --project-name test-comp up -d
+EOF
+    if [ $rc -ne 0 ]; then
+        echo "Unexpected commands executed (see the above diff), logs follow:"
+        cat "${WORKDIR}/docker-compose.log"
+        return $rc
+    fi
+
+    if ! [ -f "${PERSISTENT_DIR}/current/manifests/docker-compose.yml" ]; then
+        echo "New composition doesn't exist at the expected location"
+        return 1
+    fi
+    if ! [ -f "${PERSISTENT_DIR}/current/image_ids" ]; then
+        echo "Image IDs file doesn't exist at the expected location"
+        return 1
+    fi
+    if [ -d "${PERSISTENT_DIR}/new" ]; then
+        echo "'new' directory exists after commit"
+        return 1
+    fi
+
+    return $rc
+}
+tests+=(test_artifact_install_commit)
+
+test_artifact_install_rollback_cleanup() {
+    local rc=0
+
+    prepare_config
+    prepare_expected_file_tree
+    prepare_docker_mock
+
+    "${SRCDIR}/docker-compose" ArtifactInstall "${WORKDIR}/artifact" > "${WORKDIR}/docker-compose.log" 2>&1 || rc=$?
+    if [ $rc -ne 0 ]; then
+        echo "ArtifactInstall failed (exit code $rc), logs follow:"
+        cat "${WORKDIR}/docker-compose.log"
+        return $rc
+    fi
+    "${SRCDIR}/docker-compose" ArtifactRollback "${WORKDIR}/artifact" >> "${WORKDIR}/docker-compose.log" 2>&1 || rc=$?
+    if [ $rc -ne 0 ]; then
+        echo "ArtifactRollback failed (exit code $rc), logs follow:"
+        cat "${WORKDIR}/docker-compose.log"
+        return $rc
+    fi
+
+    local image_id1=$(head -n1 "${PERSISTENT_DIR}/cleanup/image_ids")
+    local image_id2=$(tail -n1 "${PERSISTENT_DIR}/cleanup/image_ids")
+    "${SRCDIR}/docker-compose" Cleanup "${WORKDIR}/artifact" >> "${WORKDIR}/docker-compose.log" 2>&1 || rc=$?
+    if [ $rc -ne 0 ]; then
+        echo "Cleanup failed (exit code $rc), logs follow:"
+        cat "${WORKDIR}/docker-compose.log"
+        return $rc
+    fi
+
+    cat << EOF | diff -u - "$CMDLINE_LOGGER_LOG_FILE" || rc=$?
+docker image load
+docker image load
+docker images --format {{json .ID}} some/lighttpd:latest
+docker images --format {{json .ID}} bad/php:oldest
+docker-compose --project-name test-comp up -d
+docker-compose --project-name test-comp down
+docker rmi $image_id1
+docker rmi $image_id2
+EOF
+    if [ $rc -ne 0 ]; then
+        echo "Unexpected commands executed (see the above diff), logs follow:"
+        cat "${WORKDIR}/docker-compose.log"
+        return $rc
+    fi
+
+    if [ -d "${PERSISTENT_DIR}/new" ]; then
+        echo "'new' directory exists after rollback and cleanup"
+        return 1
+    fi
+    if [ -d "${PERSISTENT_DIR}/current" ]; then
+        echo "'current' directory exists after rollback and cleanup"
+        return 1
+    fi
+    if [ -d "${PERSISTENT_DIR}/cleanup" ]; then
+        echo "'cleanup' directory exists after rollback"
+        return 1
+    fi
+
+    return $rc
+}
+tests+=(test_artifact_install_rollback_cleanup)
+
+test_artifact_install_commit_install_commit_cleanup() {
```

Comment:
This and the next test, the point is that they have an already installed artifact and then do the second one (the test). Consider making this explicit in the titles with something like
* `test_two_artifacts_install_commit_cleanup` / `test_two_artifacts_install_rollback_cleanup`
* `test_existing_artifact_install_commit_cleanup` / `test_existing_artifact_install_rollback_cleanup`
* or something like that